### ISSUE TITLE
[2020.09] PHP Warnings

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#8876]
Please report here any PHP Warning message obtained running the `2020.07` stable or `2020.09-dev` versions.

Follow-up to #8474

**Comment by AndyHee** [comment]
I get these warnings on `2020.07`

```
PHP Warning:  Error while sending STMT_PREPARE packet. PID=24067 in /var/www/friendica/src/Database/Database.php on line 533
PHP Warning:  Error while sending STMT_PREPARE packet. PID=24065 in /var/www/friendica/src/Database/Database.php on line 546
```

**Comment by AlfredSK** [comment]
NSFW add-on:

```
PHP Warning:  preg_match(): No ending delimiter '/' found in .../addon/nsfw/nsfw.php on line 133
```

I learned recently that one can use regexp in the NSFW add-on. I learned that because I get hundreds of PHP warnings per day. Apparently the regexp has to be enclosed with '/'.

I think some user input in the NSFW add-on triggers that warnings about the delimiter. Must be some regexp that is missing a '/'. Can we do something about it? Maybe some check whether the inserted regexp is well-formed.

**Comment by AlfredSK** [comment]
2020.09-dev PHP 7.4

```
PHP Warning:  strpos(): Empty needle in .../src/Content/PageInfo.php on line 296
```

**Comment by MrPetovan** [comment]
Reported by @annando 
```
PHP Warning:  htmlspecialchars() expects parameter 1 to be string, array given in /view/smarty3/compiled/c8703a755b03af3d98c1f809707f8e497a36a96e_2.file.filebrowser.tpl.php on line 44
```

**Comment by AlfredSK** [comment]
2020.09-dev PHP 7.4

After pulling this https://github.com/friendica/friendica/pull/8987 I get a _lot_ of these warnings now:

```
session_write_close(): Failed to write session data using user defined save handler. (session.save_path: /var/lib/php/sessions) in Unknown on line 0
```

`/var/lib/php/sessions` is my system's default session path. But for Friendica I use `redis` for caching and storing session data.

**Comment by MrPetovan** [comment]
I know why. The Redis Cache driver returns false when a key deletion command doesn't end up deleting an actual key. This is passed back to the session save handler which returns this error message. The expected behavior for returning false is an actual error, which Redis doesn't seem to support for deleting keys.

**Comment by copiis** [comment]
develop
`PHP Warning:  Error while sending STMT_PREPARE packet. PID=6906 in /friendica/src/Database/Database.php on line 546`

**Comment by cnleo** [comment]
`2020-08-26T09:08:01Z index [WARNING]: API call - not implemented {"module":"api","action":"call","query":"api\/v1\/directory?limit=1"} - {"file":"api.php","line":386,"function":"api_call","uid":"875932","process_id":611830}`

**Comment by annando** [comment]
@cnleo this is no PHP warning, but just a hint that the mentioned API call is not implemented.

**Comment by AlfredSK** [comment]
2020.09-rc:

`count(): Parameter must be an array or an object that implements Countable in .../src/Model/Mail.php on line 134`

There's also a notice with the same time stamp.

Fixed by #9233

**Comment by MrPetovan** [comment]
Closed by #9251
